source/s2n/s2n_tls_channel_handler.c

```diff
@@ -622,6 +631,15 @@ static int s_s2n_handler_process_read_message(
         (unsigned long long)downstream_window - processed);
 
     return AWS_OP_SUCCESS;
+
+shutdown_channel:
+    if (s2n_handler->delay_shutdown_scheduled) {
+        aws_channel_slot_on_handler_shutdown_complete(slot, AWS_CHANNEL_DIR_READ, shutdown_error_code, false);
```

Comment:
if the channel is shutting down with a non-zero error-code, but s2n_recv() returns 0, we'll end up passing along 0 as the error-code.

seems like, if `s2n_handler->delayed_shutdown_error_code != 0`, we should prefer to pass along THAT error-code